lib/nerves_hub_link/extensions/health/default_report.ex

```diff
@@ -30,6 +30,9 @@ defmodule NervesHubLink.Extensions.Health.DefaultReport do
           {"bad alarm term", ""}
       end
     end
+  rescue
+    error ->
+      %{NervesHubLink.AlarmReportFailed => inspect(error)}
```

Comment:
This breaks the Report spec of `%{String.t() => String.t()}`. But also, I think a simple case here would be simpler and handle situations where a potentially replaced alarm handler does not return expected result

```elixir
case :alarm_handler.get_alarms() do
  alarms when is_list(alarms) ->
    for {id, description} <- alarms, into: %{}, do: {inspect(id), inspect(description)}
  err ->
    %{"NervesHubLink.AlarmReportFailed" => inspect(err)}
end
```

There should not be a need to catch `inspect/1` calls. If that were to error, it would be a core problem in Elixir and we'd probably want to know.

Reply:
Changed to this suggestion in the latest commit, and it works fine but the dialyzer is complaining, maybe because it expects `:alarm_handler.get_alarms` to always return a list?

Reply:
I think there needs to be another way. `:alarm_handler.get_alarms/0` seems like it's only for experimenting with the default alarm handler. The default alarm handler is missing so many features (deduplication being a really important one) that perhaps it's better to let the user know that they should install a proper alarm handler if they want this information.

`{:error, :badmodule}` is a successful return value since it indicates that the default alarm handler has been upgraded to a proper one.

There's no generic way of getting the current set of alarms. The only option I can think of is to let the user specify a MFA or function ref to get them. That would also let the user filter alarms if they don't want them all being reported.

Reply:
@fhunleth with "let the user specify", would that be via nerveshub web or with a config setting for the nerves project?

Reply:
Oh, this never got merged. That was the intent:
https://github.com/nerves-hub/nerves_hub_link/pull/240

I was assuming that `:alarm_handler.get_alams/0` could be trusted under `alarmist` or other reasonable handlers.